### PR TITLE
[Auto-Logout] Implement Vault Timeout Options

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes
- **jslib**: Updated from `31a2574` to `28e3fff`
- **lock.component.ts**: Updated import. Updated constructor. 
- **settings.component.html**: Updated name dependencies. Added `vaultTimeoutAction` options.
- **settings.component.ts**: Updated import. Updated variable names to reflect `vaultTimeout..` pattern. Updated constructor. Updated necessary calls to the new `vaultTimeoutService`. Saving new selected action.
- **app.component.ts**: Updated import and added `PolicyService`.  The `PolicyService` call to `clear` was missing from (only) the desktop client log out cleanup. Updated necessary calls to the new `vaultTimeoutService`.
- **services.module.ts**: Updated imports. Added callback to log out and updated variables to use `vaultTimeoutService`.
- **messages.json**: Added necessary strings
- **main.ts**: Updated defaults for vault timeout settings
- **powerMonitor.main.ts**: Updated methods to lock/log based on saved action. Added tuple return type for evaluation.
- **misc.scss**: Added radio button styles.

## Screenshots
Screen Shot 2020-03-31 at 2 00 14 PM

### CODE REVIEW

File: src/scss/misc.scss [hunk 1]
```diff
@@ -253,6 +253,27 @@ form, .form {
         }
     }
 
+    .radio {
```
Comment (cscharf) [inline src/scss/misc.scss]: Curious, was there a reason additional Radio styling was necessary vs. the built-in bootstrap form-check/form-check-input/form-check-label styling used in other components?

File: src/app/accounts/settings.component.html [hunk 2]
```diff
@@ -8,12 +8,33 @@
...
+                                <label for="vaultTimeoutActionLogOut">
+                                    <input type="radio" name="VaultTimeoutAction" id="vaultTimeoutActionLogOut"
+                                        value="logOut" [(ngModel)]="vaultTimeoutAction"
+                                        (change)="saveVaultTimeoutOptions()">
+                                    {{'logOut' | i18n}}
+                                </label>
+                            </div>
+                            <small class="help-block">{{'vaultTimeoutActionLogOutDesc' | i18n}}</small>
                         </div>
                         <div class="form-group">
```
[the remark this answers is not in the document]
Comment (kspearrin) [reply src/app/accounts/settings.component.html:39]: You can still manually lock your vault, even if not by timeout, so it should remain.

File: src/locales/en/messages.json [hunk 3]
```diff
@@ -1279,5 +1279,17 @@
...
+    "message": "Vault Timeout Action"
+  },
+  "vaultTimeoutActionLockDesc": {
+    "message": "A locked vault requires that you re-enter your master password to access it again."
+  },
+  "vaultTimeoutActionLogOutDesc": {
+    "message": "A logged out vault requires that you re-authenticate to access it again."
+  },
+  "lock": {
+    "message": "Lock"
```
Comment (kspearrin) [inline src/locales/en/messages.json:1293]: We might need to add a description here for the translators to mention that this is a verb, not a padlock.

File: src/scss/misc.scss (hunk 1 above)
Comment (kspearrin) [reply src/scss/misc.scss:256]: @cscharf We only use bootstrap on the web vault. Other apps are completely custom styling for all elements (some conventions borred from bootstrap, however).

File: src/locales/en/messages.json (hunk 3 above)
Comment (vincentsalucci) [reply src/locales/en/messages.json:1293]: I've made a note and will revisit!

**Pull request merged**